### PR TITLE
Grouped parameters in separate files

### PR DESCRIPTION
**Pull request by sebglane**
This PR groups various parameters in separate files. This makes it easier to maintain the parameters and is necessary to implement the problem classes.

This PR is almost exclusively copy-paste.

### CODE REVIEW

File: include/rotatingMHD/basic_parameters.h [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+{
+
+  template<typename Stream, typename A>
+  void add_line(Stream  &stream, const A line);
+
+  template<typename Stream, typename A, typename B>
+  void add_line(Stream  &stream, const A first_column, const B second_column);
+
+  template<typename Stream>
+  void add_header(Stream  &stream);
```
Comment (j507) [inline include/rotatingMHD/basic_parameters.h:25]: Documentation

File: source/linear_solver_parameters.cc [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+
+#include <deal.II/base/conditional_ostream.h>
+
+namespace RMHD
+{
+
+/*!
+ * @brief Namespace containing all the structs and enum classes related
+ * to the run time parameters.
+ */
```
Comment (j507) [inline source/linear_solver_parameters.cc:18]: Documentation of namespaces, classes and methods should all be in the header files

**Review by j507 — APPROVED** [review]
Feel free to merge after adding the missing documentation

File: source/linear_solver_parameters.cc (hunk 2 above)
Comment (sebglane) [reply source/linear_solver_parameters.cc]: Fixed in 58843d9.

File: include/rotatingMHD/basic_parameters.h (hunk 1 above)
Comment (sebglane) [reply include/rotatingMHD/basic_parameters.h]: Fixed in 58843d9.

**Pull request merged**